### PR TITLE
Adds localization support for the mdx files and fix for the broken pages

### PR DESCRIPTION
**Pull request by kaleemullahdev**
This Pr 

i) Adds Localization support in mdx files (post/7-habit)
ii) Fix broken pages by using the correct data props in page components
iii) Removes unused graphql queries from non-page components
iv) Adds language json files for the blog page (7-habits)

### CODE REVIEW

**Review by romanr111 — CHANGES_REQUESTED** [review]
Please check.

File: src/@lekoarts/gatsby-theme-minimal-blog/components/homepage.tsx [hunk 1]
```diff
@@ -1,71 +1,70 @@
...
-import List from "./list"
-import useMinimalBlogConfig from "../hooks/use-minimal-blog-config"
-import useSiteMetadata from "../hooks/use-site-metadata"
-import replaceSlashes from "../utils/replaceSlashes"
-import { visuallyHidden } from "../styles/utils"
-import Seo from "./seo"
-import Hero from "../texts/hero.mdx"
-import Bottom from "../texts/bottom.mdx"
-import { useTranslation } from "gatsby-plugin-react-i18next";
+import { jsx } from 'theme-ui';
```
[the remark this answers is not in the document]
Comment (kaleemullahdev) [reply src/@lekoarts/gatsby-theme-minimal-blog/components/homepage.tsx:2]: Yes, it's possible that I've used nested keys for the tags in the 7-habit.json file.

File: src/@lekoarts/gatsby-theme-minimal-blog/components/homepage.tsx [hunk 2]
```diff
@@ -1,71 +1,70 @@
...
+        {t('hero_recent_posts')}
       </section>
-      <Title text={t("hero_recent_posts")}>
-        <Link to={replaceSlashes(`/${basePath}/${blogPath}`)}>Посмотреть все</Link>
+      <Title text={t('hero_recent_posts')}>
+        <Link to={replaceSlashes(`/${basePath}/${blogPath}`)}>
+          Посмотреть все
+        </Link>
       </Title>
       <Listing posts={posts} showTags={false} />
```
[the remark this answers is not in the document]
Comment (kaleemullahdev) [reply src/@lekoarts/gatsby-theme-minimal-blog/components/homepage.tsx:60]: Not for now. But I will look into this as well.

File: src/@lekoarts/gatsby-theme-minimal-blog/components/post.tsx [hunk 3]
```diff
@@ -1,99 +1,120 @@
...
+          mt: 3,
+          a: { color: `secondary` },
+          fontSize: [1, 1, 2],
+        }}
+      >
         <time>{post.date}</time>
         {post.tags && (
           <React.Fragment>
             {` — `}
-            <ItemTags tags={post.tags} />
```
[the remark this answers is not in the document]
Comment (kaleemullahdev) [reply src/@lekoarts/gatsby-theme-minimal-blog/components/post.tsx:57]: I'll fix this. Thanks

File: src/@lekoarts/gatsby-theme-minimal-blog/components/tag.tsx [hunk 4]
```diff
@@ -1,53 +1,71 @@
...
-      <Flex sx={{ alignItems: `center`, justifyContent: `space-between`, flexFlow: `wrap` }}>
+      <Flex
+        sx={{
+          alignItems: `center`,
+          justifyContent: `space-between`,
+          flexFlow: `wrap`,
+        }}
+      >
         <Heading as="h1" variant="styles.h1" sx={{ marginY: 2 }}>
           {pageContext.name}
```
Comment (romanr111) [inline src/@lekoarts/gatsby-theme-minimal-blog/components/tag.tsx:49]: This one should return translated version of the tag as a h1
Screenshot by Dropbox Capture

File: src/@lekoarts/gatsby-theme-minimal-blog/components/tags.tsx [hunk 5]
```diff
@@ -25,20 +32,27 @@ const Tags = ({ list }: MBTagsProps) => {
       </Heading>
```
Comment (romanr111) [inline src/@lekoarts/gatsby-theme-minimal-blog/components/tags.tsx:32]: This one should return translated version of the tag as a h1
Screenshot by Dropbox Capture

File: src/@lekoarts/gatsby-theme-minimal-blog/components/tags.tsx [hunk 6]
```diff
@@ -25,20 +32,27 @@ const Tags = ({ list }: MBTagsProps) => {
...
-  )
-}
+  );
+};
 
-export default Tags
+export default Tags;
 
-export const Head: HeadFC = () => <Seo title="Теги" />
+export const Head: HeadFC = () => <Seo title="Теги" />;
```
Comment (romanr111) [inline src/@lekoarts/gatsby-theme-minimal-blog/components/tags.tsx:58]: SEO title as well needs to be translated.

File: src/@lekoarts/gatsby-theme-minimal-blog/components/page.tsx [hunk 7]
```diff
@@ -1,26 +1,41 @@
...
+    </Layout>
+  );
+};
 
-export const Head: HeadFC<MBPageProps> = ({ data: { page } }) => <Seo title={page.title} description={page.excerpt} />
\ No newline at end of file
+export default Page;
+
+export const Head: HeadFC<MBPageProps> = ({ data: { page } }) => (
+  <Seo title={page.title} description={page.excerpt} />
```
Comment (romanr111) [inline src/@lekoarts/gatsby-theme-minimal-blog/components/page.tsx:40]: @kaleemullahdev Page metedata needs to be translated as well;
Screenshot by Dropbox Capture

File: src/@lekoarts/gatsby-theme-minimal-blog/components/post.tsx [hunk 8]
```diff
@@ -1,99 +1,117 @@
...
+export default Post;
 
 export const Head: HeadFC<MBPostProps> = ({ data: { post } }) => {
-  const firstTag = post.tags && post.tags[0]
-  const isBookReview = firstTag && firstTag.name === "Обзор книги"
+  const firstTag = post.tags && post.tags[0];
+  const isBookReview = firstTag && firstTag.name === 'Обзор книги';
 
-  const title = isBookReview ? `Краткое содержание - ${post.title}` : post.title
+  const title = isBookReview
```
Comment (romanr111) [inline src/@lekoarts/gatsby-theme-minimal-blog/components/post.tsx:109]: Post metadata needs to be translated as well.
Screenshot by Dropbox Capture
Screenshot by Dropbox Capture

File: content/posts/7-habits/index.mdx [hunk 9]
```diff
@@ -24,148 +28,289 @@ banner: /images/seven-habits-of-highly-effecitve-people-book.jpeg
...
 `}</style>
 
- ## Суть книги в двух словах:
+<Translation
+  id="main_heading"
+  tag="h2"
+  ns={props.data.post.ns}
+/>
+
+<Translation
```
[the remark this answers is not in the document]
Comment (romanr111) [reply content/posts/7-habits/index.mdx:44]: Here I was able to create PR into your branch.
https://github.com/kaleemullahdev/romanrud-blog/pull/1

**Comment by kaleemullahdev** [comment]
Great work @romanr111. I have one small suggestion. Instead of directly passing the markdown content to the ReactMarkdown component, consider utilizing the children prop. This will make the component more flexible, as it will allow us to customize the content using component props in the future. While this change is not necessary at the moment, it's something I'm considering for future enhancements. Thank you for your understanding.

**Review by romanr111 — APPROVED** [review]
Looks great!

**Pull request merged**